samples/Azure.ResourceManager.Sample/Generated/LongRunningOperation/SampleArmOperationOfT.cs

```diff
@@ -43,7 +51,7 @@ internal SampleArmOperation(IOperationSource<T> source, ClientDiagnostics client
 #pragma warning restore CA1822
 
         /// <inheritdoc />
-        public override T Value => _operation.Value;
+        public override T Value => !HasValue && _source.SupportInterimStatus ? _interimValue ??= _source.CreateResult(_interimResponse, default) : _operation.Value;
```

Comment:
When I thought about this change I thought we might make it directly in the base classes.  The feature itself should default to false but if try affect how [`Value` works here](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/Shared/OperationInternalOfT.cs#L157).

Off the top of my head if we add an optional param to `OperationInternal<T>` ctor and each `[RP]ArmOperation<T>` then each LRO method would know if this should be true and pass that in when constructing the Operation object otherwise the calling code would remain unchanged.

This would eliminate the need for the new interface `IArmOperationSource` and it allows everyone to take advantage of the functionality if they want to opt in.

Reply:
Added an optional parameter to `OperationInternal<T>` in https://github.com/Azure/azure-sdk-for-net/pull/29937